web/src/components/CompareCard.vue

```diff
@@ -1,46 +1,242 @@
+      cousins.classList[op]("hovering");
+    }
+  }
 
-  get codeRight(): string {
-    return this.intersection.rightFile.content;
+  onHoverEnterHandler(sideId: string, blockClasses: Array<string>, element: HTMLElement): void {
+    this.lastHovered.side = sideId;
+    this.lastHovered.blockClasses = blockClasses;
+    this.addClassesToSiblingsAndCousins("add", sideId, blockClasses[0]);
   }
 
-  get codeLeft(): string {
-    return this.intersection.leftFile.content;
+  onHoverExitHandler(sideId: string, blockClasses: Array<string>, element: HTMLElement): void {
+    this.addClassesToSiblingsAndCousins("remove", sideId, blockClasses[0]);
+  }
+
+  selectionClickEventHandler(sideId: string, blockClasses: Array<string>): void {
+    // if the there is nothing that was last clicked, or a different block from last time is clicked initialize the
+    // values
+    if (this.lastClicked.blockClasses === undefined || !(sideId === this.lastClicked.side &&
+      this.lastClicked.blockClasses.sort().toString() === blockClasses.sort().toString())) {
+      this.blockClickCount = 1;
+      this.lastClicked.side = sideId;
+      this.lastClicked.blockClasses = blockClasses;
+    }
+
+    const map = this.sideMap.get(sideId)!;
+    let id = this.lastClicked.blockClasses[this.currentBlockClassIndex];
+    let blocks = map.get(id)!;
+    // cycles through all the possible combination for the current block
+    if (this.blockClickCount === blocks.length) {
+      this.blockClickCount = 1;
+      this.currentBlockClassIndex = (this.currentBlockClassIndex + 1) % this.lastClicked.blockClasses.length;
+      id = this.lastClicked.blockClasses[this.currentBlockClassIndex];
+      blocks = map.get(id)!;
+    } else {
+      this.blockClickCount += 1;
+    }
+
+    const other = blocks.shift() as string;
+    blocks.push(other);
+
+    const visibleElements = document.querySelectorAll(".marked-code[class~=visible]") as NodeListOf<HTMLElement>;
+    for (const visibleElement of visibleElements) {
+      visibleElement.classList.remove("visible");
+    }
+
+    const firstBlocks = document.querySelectorAll(`#${sideId} .${id}`) as NodeListOf<HTMLElement>;
+    const secondBlocks = document.querySelectorAll(`pre:not(#${sideId}) .${other}`) as NodeListOf<HTMLElement>;
+    firstBlocks.forEach(val => val.classList.add("visible"));
+    secondBlocks.forEach(val => val.classList.add("visible"));
+    firstBlocks[0].scrollIntoView({ behavior: "smooth", block: "center" });
+    secondBlocks[0].scrollIntoView({ behavior: "smooth", block: "center" });
+  }
+
+  get leftSelection(): Array<Selection> {
+    return this.diff.hunks.map(fragment => fragment.left);
+  }
+
+  get rightSelection(): Array<Selection> {
+    return this.diff.hunks.map(fragment => fragment.right);
+  }
+
+  extractRowCol(value: string): [number, number] {
+    const matches = /([0-9]*)-([0-9]*)-[0-9]*-[0-9]*$/m.exec(value) as RegExpExecArray;
```

Comment:
It would not as the start of the id also contains a `-`.

Reply:
```suggestion
    const row, col = value.split('-').slice(2, 4);
```
For `-1-2-3-4`, this would yield `2` and `3` for row and col. This less complex than a regex in my opinion.